### PR TITLE
feat: LT-1328 - Open licence refactor

### PR DESCRIPTION
**Pull request by jfaracik**
* Removed fields from open licence applications
* Add deprecation warning to get_string
* Add extra CircleCI step for ensuring that migrations are made before committing.
image
* Update pre-commit hook
* Add pyproject.toml for consistent formatting
* Add unittest-xml-reporting to give XML reports on test passes/failures
image
Useful for CircleCI!

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -62,14 +62,18 @@ jobs:
...
+
       # Linting: dodgy, mccabe, pep8, profile-validator, pyflakes, pylint
       - run:
           name: Run Linting
           command: pipenv run prospector
 
       - run:
           name: check code formatting
-          command: pipenv run black . --line-length 120 --check --diff
+          command: pipenv run black . --check --diff
```
Comment (markm5A8Z9F43913A) [inline .circleci/config.yml]: Why are we getting rid of line length of 120 characters?

Comment (jfaracik) [reply .circleci/config.yml]: Its now in pyproject.toml https://black.readthedocs.io/en/stable/pyproject_toml.html

File: .circleci/config.yml [hunk 2]
```diff
@@ -79,11 +83,15 @@ jobs:
...
+      - store_test_results:
+          path: test-results
 
       # Code Coverage
       - run:
           name: Run Code Coverage
-          command: pipenv run ./manage.py coverage --threshold=80 --html=False
+          command: pipenv run ./manage.py coverage --threshold=80
+      - store_artifacts:
+          path: htmlcov
```
Comment (BenAndrew-Kainos) [inline .circleci/config.yml]: What's this for?

File: goodstype/serializers.py [hunk 3]
```diff
@@ -12,7 +12,7 @@
 
 
 class GoodsTypeSerializer(serializers.ModelSerializer):
-    description = serializers.CharField(max_length=280)
+    description = serializers.CharField(max_length=2000)
```
Comment (BenAndrew-Kainos) [inline goodstype/serializers.py]: As this length is reused in multiple places could we define a constant for it (and maybe for all char lengths)

File: docs/linting.md [hunk 4]
```diff
@@ -0,0 +1,4 @@
+We use Black to format our code.
+
+To run Black:
+* `black .`
```
Comment (efahy) [inline docs/linting.md]: I think a little more detail on how to run black is in order, at least a reference to the 120 char line length

File: goodstype/serializers.py [hunk 5]
```diff
@@ -5,20 +5,21 @@
...
+from goodstype.constants import DESCRIPTION_MAX_LENGTH
 from goodstype.document.models import GoodsTypeDocument
 from goodstype.models import GoodsType
 from static.countries.models import Country
 from static.countries.serializers import CountrySerializer
 
 
 class GoodsTypeSerializer(serializers.ModelSerializer):
-    description = serializers.CharField(max_length=280)
+    description = serializers.CharField(max_length=DESCRIPTION_MAX_LENGTH)
```
Comment (efahy) [inline goodstype/serializers.py]: Max length for standard goods is still `280` - should you not consider updating it at the same time?

File: static/management/commands/seedgovuser.py [hunk 6]
```diff
@@ -32,6 +32,7 @@ def operation(self, *args, **options):
                 Permissions.MANAGE_TEAM_ADVICE,
                 Permissions.REVIEW_GOODS,
                 Permissions.ADMINISTER_ROLES,
+                Permissions.CONFIRM_OWN_ADVICE,
```
Comment (efahy) [inline static/management/commands/seedgovuser.py]: Why are you adding this in your PR, it's not part of your story?

File: conf/settings.py [hunk 7]
```diff
@@ -11,7 +11,7 @@
 if os.path.exists(ENV_FILE):
     Env.read_env(ENV_FILE)
 
-env = Env(ALLOWED_HOSTS=(str, ""), DEBUG=(bool, False), LOG_LEVEL=(str, "INFO"), BACKGROUND_TASK_ENABLED=(bool, False),)
+env = Env(ALLOWED_HOSTS=(str, ""), DEBUG=(bool, False), LOG_LEVEL=(str, "INFO"), BACKGROUND_TASK_ENABLED=(bool, False))
```
Comment (efahy) [inline conf/settings.py]: Was there a gap in the original black API implementation by Craig that resulted in these formatting changes?  Is there something we need to fix with his config retrospectively?

File: goodstype/serializers.py (hunk 5 above)
Comment (jfaracik) [reply goodstype/serializers.py]: That's not in the story, I will ask PO tomorrow about it if you want.

File: static/management/commands/seedgovuser.py (hunk 6 above)
Comment (jfaracik) [reply static/management/commands/seedgovuser.py]: Tests were failing as a result of this missing :/

File: docs/linting.md (hunk 4 above)
Comment (jfaracik) [reply docs/linting.md]: Line length is done automatically, `black .` is all you need to run.

File: conf/settings.py (hunk 7 above)
Comment (jfaracik) [reply conf/settings.py]: Not too sure on this one

**Pull request merged**